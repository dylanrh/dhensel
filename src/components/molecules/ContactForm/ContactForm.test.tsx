import { ContactForm } from '../ContactForm';

import { render } from '@testing-library/react';

jest.spyOn(console, 'error').mockImplementation(jest.fn);

describe('ContactForm', () => {
	it('renders correctly', () => {
		const { container } = render(<ContactForm />);
		expect(container).toMatchSnapshot();
	});
});
