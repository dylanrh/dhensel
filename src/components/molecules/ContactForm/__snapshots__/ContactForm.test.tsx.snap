// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm renders correctly 1`] = `
<div>
  <form
    action="#"
    class="mt-4"
    role="form"
  >
    <div
      class="glass-light dark:glass-dark mb-6 rounded-xl px-8 py-6 drop-shadow-lg"
    >
      <label
        class="mb-2 mt-6 flex flex-col font-bold"
        for="firstName"
      >
        First Name
        <input
          class="my-2 rounded-md px-4 py-2 ring-1 dark:bg-transparent ring-grey-400 dark:ring-slate-500"
          id="firstName"
          name="firstName"
          placeholder="John"
          type="text"
          value=""
        />
        <div
          class="font-sm font-normal text-red-600"
        />
      </label>
      <label
        class="mb-2 mt-6 flex flex-col font-bold"
        for="firstName"
      >
        Last Name
        <input
          class="my-2 rounded-md px-4 py-2 ring-1 dark:bg-transparent ring-grey-400 dark:ring-slate-500"
          id="lastName"
          name="lastName"
          placeholder="Doe"
          type="text"
          value=""
        />
        <div
          class="font-sm font-normal text-red-600"
        />
      </label>
      <label
        class="mb-2 mt-6 flex flex-col font-bold"
        for="firstName"
      >
        Company
        <input
          class="my-2 rounded-md px-4 py-2 ring-1 dark:bg-transparent ring-grey-400 dark:ring-slate-500"
          id="company"
          name="company"
          type="text"
          value=""
        />
        <div
          class="font-sm font-normal text-red-600"
        />
      </label>
      <label
        class="mb-2 mt-6 flex flex-col font-bold"
        for="firstName"
      >
        Email Address
        <input
          class="my-2 rounded-md px-4 py-2 ring-1 dark:bg-transparent ring-grey-400 dark:ring-slate-500"
          id="email"
          name="email"
          placeholder="johndoe@gmail.com"
          type="email"
          value=""
        />
        <div
          class="font-sm font-normal text-red-600"
        />
      </label>
      <label
        class="mb-2 mt-6 flex flex-col font-bold"
        for="subject"
      >
        Subject
        <select
          class="my-2 rounded-md px-4 py-2 ring-1 dark:bg-transparent ring-grey-400 dark:ring-slate-500"
          id="subject"
          name="subject"
        >
          <option
            value="none"
          >
            Please select
          </option>
          <option
            value="I need a website / app developed"
          >
            I need a website / app developed
          </option>
          <option
            value="Feedback on this website"
          >
            Feedback on this website
          </option>
          <option
            value="General Inquiry"
          >
            General Inquiry
          </option>
          <option
            value="I am a recruiter and want to hire you"
          >
            I am a recruiter and want to hire you
          </option>
          <option
            value="Other"
          >
            Other
          </option>
        </select>
        <div
          class="font-sm font-normal text-red-600"
        />
      </label>
      <label
        class="mb-2 mt-6 flex flex-col font-bold"
        for="firstName"
      >
        Message
        <textarea
          class="my-2 rounded-md px-4 py-2 ring-1 dark:bg-transparent ring-grey-400 dark:ring-slate-500"
          id="message"
          name="message"
          rows="5"
        />
        <div
          class="font-sm font-normal text-red-600"
        />
      </label>
      <div
        class="mt-6"
      >
        <button
          class="rounded-md bg-blue-700 hover:bg-blue-600 active:bg-blue-800 px-3.5 py-2.5 text-sm font-semibold text-white shadow-sm focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-600 transition-all group"
          type="submit"
        >
          Send Message
        </button>
      </div>
    </div>
  </form>
</div>
`;
